contentScript.js

```diff
@@ -0,0 +1,149 @@
+(() => {
+
+	let ytLeftControls, ytPlayer;
+	let currentVid = null;  // id of the currently playing video
+	let lastVid = null;   // id of the last handled video
+	let BmsList = [];  // list of the video bookmarks
+    
+		/* Fetch all bookmarks for current video */
+		const fetchBookmarks = () => {
+			return new Promise(resolve => {
+				// get all bookmarks for current video and create bookmark elements
+				chrome.storage.sync.get([ currentVid ], (obj) => {
+					resolve(obj[currentVid] ? JSON.parse(obj[currentVid]) : [])
+				});
+			});
+		}
+	
+		/* Clear out all previous bookmarks */
+		const delOldBms = () => {
+			// remove bookmarks from previous video
+			const oldBms = document.getElementsByClassName('bookmark');
+			while(oldBms[0]) { oldBms[0].parentNode.removeChild(oldBms[0]);}
+		}
+	
+		/* get current video's duration */
+		const getVideoDuration = () => {
+			if (ytPlayer) { return parseInt(ytPlayer.duration);}
+			return null;
+		}
+
+			/* Handle request to add new bookmark */
+			const addNewBmEventHandler = () => {
+	
+		let currentTime = null;
+		if (ytPlayer) {
+		 currentTime = parseInt(ytPlayer.currentTime);
+		}
+
+		if (currentTime) {
+
+			const newBm = { time: currentTime, desc: 'Bookmark at ' + getTime(currentTime) };
+
+			fetchBookmarks().then(obj => {
+				
+				BmsList = obj;
+				BmsList.push(newBm);
+				BmsList = BmsList.sort((a, b) => ( a.time - b.time));
+				chrome.storage.sync.set({[currentVid]: JSON.stringify(BmsList)});
+			});
+		}
+	}
+
+		/* this checks for whenever a new video is loaded or the current tab is reloaded */
+		const newVideoLoaded = (refreshed) => {
+
+			delOldBms();
+			BmsList = [];
+			ytLeftControls = document.getElementsByClassName('ytp-left-controls')[0];
+			ytPlayer = document.getElementsByClassName("html5-main-video")[0];
+		
+	
+			const fetchBmsPromise = fetchBookmarks();
+	
+			const videoBmTimePromise = new Promise((resolve) => {
+				const getCurrentTime = () => {
+					ytPlayer.removeEventListener('playing', getCurrentTime);
+					resolve(getVideoDuration());
+				}
+	
+				if (refreshed) {
+					resolve(getVideoDuration());
+				} else {
+					ytPlayer.addEventListener('playing', getCurrentTime);
+				}
```

Comment:
I typically try to add early exit patturns and also avoid using an else at all costs like this

`
if (refreshed) {
        resolve(getVideoDuration());
	return;
}

ytPlayer.addEventListener('playing', getCurrentTime);

`